Eigen/src/Core/arch/GPU/Half.h

```diff
@@ -49,14 +49,15 @@ struct half;
 
 namespace half_impl {
 
-#if !defined(EIGEN_HAS_HIP_FP16)
+#if !defined(EIGEN_HAS_GPU_FP16)
 // Make our own __half_raw definition that is similar to CUDA's.
 struct __half_raw {
   EIGEN_DEVICE_FUNC __half_raw() : x(0) {}
   explicit EIGEN_DEVICE_FUNC __half_raw(unsigned short raw) : x(raw) {}
   unsigned short x;
 };
-#elif defined(EIGEN_HAS_OLD_HIP_FP16)
+#elif defined(EIGEN_HAS_HIP_FP16)
+ #if defined(EIGEN_HAS_OLD_HIP_FP16)
```

Comment:
where is `EIGEN_HAS_OLD_HIP_FP16` defined?

Reply:
Eigen/Core

Reply:
i fail to identify the place, could you post a pointer here?

Reply:
https://github.com/ROCmSoftwarePlatform/eigen-upstream/blob/PR_fixes/Eigen/Core#L364